index.js

```diff
@@ -251,6 +251,32 @@ const execa = (command, args, options) => {
 		return Promise.reject(error);
 	}
 
+	const originalKill = spawned.kill.bind(spawned);
+	spawned.kill = (signal, options) => {
+		const killResult = originalKill(signal);
+		if (signal === undefined ||
+			(typeof signal === 'string' && signal.toUpperCase() === 'SIGTERM') ||
+			(typeof signal === 'number' && signal === os.constants.signals.SIGTERM)) {
```

Comment:
This could be made more readable by extracting the condition into a `isSigterm` variable.

Reply:
Can `signal` actually be a number for `cp.kill()`? Not according to the Node.js docs, at least: https://nodejs.org/api/child_process.html#child_process_subprocess_kill_signal

Reply:
`signal` can be a `number`: 
https://github.com/nodejs/node/blob/908292cf1f551c614a733d858528ffb13fb3a524/lib/internal/child_process.js#L435
https://github.com/nodejs/node/blob/908292cf1f551c614a733d858528ffb13fb3a524/lib/internal/util.js#L220

But it's undocumented (intentionally or not) it seems. I have [filled an issue in Node.js](https://github.com/nodejs/node/issues/27750)